astronomer/providers/amazon/aws/extractors/redshift.py

```diff
@@ -0,0 +1,235 @@
+        """
+        self.log.debug("extract_on_complete(%s)", task_instance)
+        job_facets = {"sql": SqlJobFacet(self.operator.sql)}
+
+        self.log.debug("Sending SQL to parser: %s", self.operator.sql)
+        sql_meta: Optional[SqlMeta] = parse(self.operator.sql, self.default_schema)
+        self.log.debug("Got meta %s", sql_meta)
+        try:
+            redshift_job_id = self._get_xcom_redshift_job_id(task_instance)
+            if redshift_job_id is None:
+                raise Exception("Xcom could not resolve Redshift job id. Job may have failed.")
+        except Exception:
+            return TaskMetadata(
+                name=get_job_name(task=self.operator),
+                run_facets={},
+                job_facets=job_facets,
+            )
+
+        client = self.operator.hook.conn
+
+        redshift_details = [
+            "database",
+            "cluster_identifier",
+            "db_user",
+            "secret_arn",
+            "region",
+        ]
+
+        connection_details = {detail: getattr(self.operator, detail) for detail in redshift_details}
+
+        stats = RedshiftDataDatasetsProvider(client=client, connection_details=connection_details).get_facets(
+            job_id=redshift_job_id[0],
+            inputs=sql_meta.in_tables if sql_meta else [],
+            outputs=sql_meta.out_tables if sql_meta else [],
+        )
+
+        return TaskMetadata(
+            name=get_job_name(task=self.operator),
+            inputs=[ds.to_openlineage_dataset() for ds in stats.inputs],
+            outputs=[ds.to_openlineage_dataset() for ds in stats.output],
+            run_facets=stats.run_facets,
+            job_facets={"sql": SqlJobFacet(self.operator.sql)},
+        )
+
+    def _get_xcom_redshift_job_id(self, task_instance: TaskInstance) -> List[str]:
+        """Get query ids from XCOM"""
+        redshift_job_id: List[str]
+        redshift_job_id = task_instance.xcom_pull(task_ids=task_instance.task_id, key="query_ids")
+
+        self.log.debug("redshift job id: %s", redshift_job_id)
+        return redshift_job_id
+
+
+# To be removed once redshift extractors are released in openlineage
+@attr.s
+class ErrorMessageRunFacet(BaseFacet):
+    """This facet represents an error message that was the result of a job run"""
+
+    message: str = attr.ib()
+    programmingLanguage: str = attr.ib()
+    stackTrace: Optional[str] = attr.ib(default=None)
+
+    @staticmethod
+    def _get_schema() -> str:
+        return SCHEMA_URI + "#/definitions/ErrorMessageRunFacet"
+
+
+# To be removed once redshift extractors are released in openlineage
+@attr.s
+class RedshiftFacets:
+    """This facet represents a RedshiftFacets"""
+
+    run_facets: Dict[str, BaseFacet] = attr.ib()
+    inputs: List[Dataset] = attr.ib()
+    output: List[Dataset] = attr.ib()
+
+
+# To be removed once redshift extractors are released in openlineage
+class RedshiftDataDatasetsProvider:
+    """RedshiftDataDatasetsProvider"""
```

Comment:
Need proper Doc string

Reply:
Removed this as ``openlineage-airflow>=0.12.0`` got released. We don't need this code anymore in our repo.